### PR TITLE
Folder restructure

### PR DESCRIPTION
**Pull request by glimchb**
split to several packages:
- frontend
- middleend
- backend

This will help importing this library from other projects

TODOs:
- remove some code dup in tests
- split to more files like nvme vs virtio

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

File: pkg/frontend/frontend.go [hunk 1]
```diff
@@ -11,27 +12,36 @@ import (
...
 
+// Server represents the Server object
+type Server struct {
+	pb.UnimplementedFrontendNvmeServiceServer
+	pb.UnimplementedFrontendVirtioBlkServiceServer
+	pb.UnimplementedFrontendVirtioScsiServiceServer
+}
+
 // ////////////////////////////////////////////////////////
 var subsystems = map[string]*pb.NVMeSubsystem{}
```
[the remark this answers is not in the document]
Comment (glimchb) [reply pkg/frontend/frontend.go:31]: Agree

File: pkg/backend/backend_test.go [hunk 2]
```diff
@@ -1,26 +1,114 @@
...
+	"context"
 	"fmt"
 	"log"
 	"net"
+	"os"
 	"reflect"
+	"strings"
 	"testing"
 
+	"google.golang.org/protobuf/proto"
```
Comment (seroyer) [inline pkg/backend/backend_test.go:18]: nit, but I'd organize this with the other protobuf import below

**Review by sandersms — APPROVED** [review]
Changes look good.  Other comments will be resolved at later time.

File: pkg/backend/backend.go [hunk 3]
```diff
@@ -13,18 +13,27 @@ import (
...
 	pb "github.com/opiproject/opi-api/storage/v1alpha1/gen/go"
+	"github.com/opiproject/opi-spdk-bridge/pkg/server"
+
 	"github.com/ulule/deepcopier"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
+// Server represents the Server object
```
[the remark this answers is not in the document]
Comment (glimchb) [reply pkg/backend/backend.go:24]: Ok. Will do

File: cmd/main.go [hunk 4]
```diff
@@ -27,13 +29,13 @@ func main() {
...
 	s := grpc.NewServer()
 
-	pb.RegisterFrontendNvmeServiceServer(s, &server.Server{})
-	pb.RegisterNVMfRemoteControllerServiceServer(s, &server.Server{})
-	pb.RegisterFrontendVirtioBlkServiceServer(s, &server.Server{})
-	pb.RegisterFrontendVirtioScsiServiceServer(s, &server.Server{})
-	pb.RegisterNullDebugServiceServer(s, &server.Server{})
-	pb.RegisterAioControllerServiceServer(s, &server.Server{})
-	pb.RegisterMiddleendServiceServer(s, &server.Server{})
+	pb.RegisterFrontendNvmeServiceServer(s, &frontend.Server{})
```
[the remark this answers is not in the document]
Comment (glimchb) [reply cmd/main.go:32]: Hmm good question

**Review by seroyer — APPROVED** [review]

File: pkg/backend/backend_test.go (hunk 2 above)
Comment (glimchb) [reply pkg/backend/backend_test.go:18]: I'll run `goimport` tool to auto-sort this

**Comment by glimchb** [comment]
@artek-koltun I will merge this one and address your comments in a new PR that I already have pending.
Just want to sure I can import correctly this split from other repos and if not - fix it together

**Pull request merged**